core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2792,27 +2792,26 @@ public static Collection multisetUnionAll(Collection collection1,
    * Function that, given a certain List containing single-item structs (i.e. arrays / lists with
    * a single item), builds an Enumerable that returns those single items inside the structs.
    */
-  public static Function1<Object, Enumerable<Comparable>> flatList() {
-    return inputObject -> {
-      final List list = (List) inputObject;
-      final Enumerator<List<Object>> enumerator = Linq4j.enumerator(list);
-      return new AbstractEnumerable<Comparable>() {
-        @Override public Enumerator<Comparable> enumerator() {
-          return new Enumerator<Comparable>() {
+  public static Function1<List<Object>, Enumerable<Object>> flatList() {
+    return inputList -> {
+      final Enumerator<Object> enumerator = Linq4j.enumerator(inputList);
+      return new AbstractEnumerable<Object>() {
+        @Override public Enumerator<Object> enumerator() {
+          return new Enumerator<Object>() {
 
             @Override public boolean moveNext() {
               return enumerator.moveNext();
             }
 
-            @Override public Comparable current() {
+            @Override public Object current() {
               final Object element = enumerator.current();
-              final Comparable comparable;
+              final Object current;
               if (element.getClass().isArray()) {
-                comparable = (Comparable) ((Object[]) element)[0];
+                current = ((Object[]) element)[0];
               } else {
-                comparable = (Comparable) ((List) element).get(0);
+                current = ((List) element).get(0);
               }
```

Comment:
What do you think of reusing `org.apache.calcite.runtime.SqlFunctions#structAccess` here?

Reply:
Done.